aptos-move/move-examples/data_structures/sources/big_vector.move

```diff
@@ -5,226 +5,231 @@ module aptos_std::big_vector {
         // and append the last_val to the bucket then swap the last bucket val back
-        let bucket = table_with_length::borrow_mut(&mut v.buckets, index.bucket_index);
+        let bucket = table_with_length::borrow_mut(&mut v.buckets, i / v.bucket_size);
         let bucket_len = vector::length(bucket);
-        let val = vector::swap_remove(bucket, index.vec_index);
+        let val = vector::swap_remove(bucket, i % v.bucket_size);
         vector::push_back(bucket, last_val);
-        vector::swap(bucket, index.vec_index, bucket_len - 1);
+        vector::swap(bucket, i % v.bucket_size, bucket_len - 1);
         val
     }
 
-    /// Return true if `val` is in the vector `v`.
-    public fun contains<T>(v: &BigVector<T>, val: &T): bool {
-        if (is_empty(v)) return false;
-        let (exist, _) = index_of(v, val);
-        exist
+    /// Swap the elements at the i'th and j'th indices in the vector v. Will abort if either of i or j are out of bounds
+    /// for v.
+    public fun swap<T>(v: &mut BigVector<T>, i: u64, j: u64) {
+        assert!(i < length(v) && j < length(v), error::invalid_argument(EINDEX_OUT_OF_BOUNDS));
+        let i_bucket_index = i / v.bucket_size;
+        let j_bucket_index = j / v.bucket_size;
+        let i_vector_index = i % v.bucket_size;
+        let j_vector_index = j % v.bucket_size;
+        if (i_bucket_index == j_bucket_index) {
+            vector::swap(table_with_length::borrow_mut(&mut v.buckets, i_bucket_index), i_vector_index, j_vector_index);
+            return
+        };
+        let bucket_i = table_with_length::remove(&mut v.buckets, i_bucket_index);
+        let bucket_j = table_with_length::remove(&mut v.buckets, j_bucket_index);
+        let element_i = vector::swap_remove(&mut bucket_i, i_vector_index);
+        let element_j = vector::swap_remove(&mut bucket_j, j_vector_index);
+        vector::push_back(&mut bucket_i, element_j);
+        vector::push_back(&mut bucket_j, element_i);
+        let last_index_in_bucket_i = vector::length(&bucket_i) - 1;
+        let last_index_in_bucket_j = vector::length(&bucket_j) - 1;
+        vector::swap(&mut bucket_i, i_vector_index, last_index_in_bucket_i);
+        vector::swap(&mut bucket_j, j_vector_index, last_index_in_bucket_j);
+        table_with_length::add(&mut v.buckets, i_bucket_index, bucket_i);
+        table_with_length::add(&mut v.buckets, j_bucket_index, bucket_j);
     }
 
-    /// Return `(true, i)` if `val` is in the vector `v` at index `i`.
-    /// Otherwise, returns `(false, 0)`.
+    /// Reverse the order of the elements in the vector v in-place.
+    /// disclaimer: this function is costly. Use it at your own discretion.
+    public fun reverse<T>(v: &mut BigVector<T>) {
+        let len = length(v);
+        let half_len = len / 2;
+        let k = 0;
+        while (k < half_len) {
+            swap(v, k, len - 1 - k);
+            k = k + 1;
+        }
```

Comment:
can be more efficient using `vector::reverse()`? (because it uses native `swap`)

Reply:
discussed offline. hold it on.